diff --git a/kernel/manager/manager_sign.h b/kernel/manager/manager_sign.h
index 87e78338..c7fee32e 100644
--- a/kernel/manager/manager_sign.h
+++ b/kernel/manager/manager_sign.h
@@ -25,6 +25,10 @@
 #define EXPECTED_SIZE_KOWX712 0x375
 #define EXPECTED_HASH_KOWX712 "484fcba6e6c43b1fb09700633bf2fb4758f13cb0b2f4457b80d075084b26c588"
 
+// KernelSU-Next/KernelSU-Next
+#define EXPECTED_SIZE_KSUN 0x3e6
+#define EXPECTED_HASH_KSUN "79e590113c4c4c0c222978e413a5faa801666957b1212a328e46c00c69821bf7"
+
 typedef struct {
     unsigned size;
     const char *sha256;
